packages/ui-kit/src/helpers/useCombinedRefsCallback.ts

```diff
@@ -0,0 +1,53 @@
+import { useState, useCallback, Ref, MutableRefObject } from 'react'
+import { useCombinedRefs } from './useCombinedRefs'
+
+/**
+ * Custom React hook that creates a combined ref and provides a callback to set it.
+ * It utilizes `useCombinedRefs` to merge `forwardedRef` and `innerRef` into a single ref (`combinedRef`).
+ * Additionally, it manages a state (`componentContainer`) to store the DOM element associated with the ref.
+ * The `setRef` callback is used to update both the `componentContainer` state and the `combinedRef`.
+ * This hook is useful for components that need to expose their DOM node to parent components while maintaining an internal ref.
+ *
+ * @param {Ref<HTMLElement>} forwardedRef - The forwarded ref.
+ * @param {Ref<HTMLElement>} innerRef - The internal ref.
+ * @returns An object containing `componentContainer`, `combinedRef`, and `setRef`.
+ */
+export const useCombinedRefsCallback = ({
+  forwardedRef,
+  innerRef
+}: {
+  forwardedRef: Ref<HTMLElement>
+  innerRef: Ref<HTMLElement>
```

Comment:
Those are more likely the correct types

```suggestion
  forwardedRef: ForwardedRef<HTMLElement>
  innerRef: MutableRefObject<HTMLElement> | RefCallback<HTMLElement>
```

Reply:
Yeah, that might work better, thanks! It also requires the use of generics.